embulk-jruby-extension/src/main/java/org/jruby/util/RubyDateParser.java

```diff
@@ -0,0 +1,608 @@
+package org.jruby.util;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jruby.Ruby;
+import org.jruby.RubyString;
+import org.jruby.runtime.ThreadContext;
+
+import static org.jruby.util.RubyDateFormatter.Format;
+import static org.jruby.util.RubyDateFormatter.Token;
+
+public class RubyDateParser
+{
+    private static String[] dayNames = new String[] {
+            "Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday",
+            "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"
+    };
+
+    private static String[] monNames = new String[] {
+            "January", "February", "March", "April", "May", "June", "July", "August", "September",
+            "October", "November", "December", "Jan", "Feb", "Mar", "Apr", "May", "Jun",
+            "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
+    };
+
+    private static String[] meridNames = new String[] {
+            "am", "pm", "a.m.", "p.m."
+    };
+
+    private static final Pattern ZONE_PARSE_REGEX = Pattern.compile("\\A(" +
+                    "(?:gmt|utc?)?[-+]\\d+(?:[,.:]\\d+(?::\\d+)?)?" +
+                    "|(?-i:[[\\p{Alpha}].\\s]+)(?:standard|daylight)\\s+time\\b" +
+                    "|(?-i:[[\\p{Alpha}]]+)(?:\\s+dst)?\\b" +
+                    ")",
+            Pattern.CASE_INSENSITIVE);
+
+    private int matchAtPatterns(String text, int pos, String[] patterns)
+    {
+        int patIndex = -1;
+        for (int i = 0; i < patterns.length; i++) {
+            String pattern = patterns[i];
+            int len = pattern.length();
+            try {
+                if (pattern.equalsIgnoreCase(text.substring(pos, pos + len))) { // IndexOutOfBounds
+                    patIndex = i;
+                    break;
+                }
+            } catch (IndexOutOfBoundsException e) {
+                // ignorable error
+            }
+        }
+        return patIndex;
+    }
+
+    private static boolean isValidRange(long v, int lower, int upper)
+    {
+        return lower <= v && v <= upper;
+    }
+
+    private static boolean isSpace(char c)
+    {
+        return c == ' ' || c == '\t' || c == '\n' ||
+                c == '\u000b' || c == '\f' || c == '\r';
+    }
```

Comment:
commented the link of the whitespace declaration: https://github.com/embulk/embulk/commit/cb49a8a